### DISCUSSION TITLE
How to add parameters to events when using getPlanFromEvents?

### DISCUSSION THREAD

**Discussion by satyasinha** [statelyai/xstate#2904] [Q&A]
Hello,

Huge fan of xstate so far, thanks for all the work that's been done here!

Been using the @xstate/test package for some testing and noticed an odd behaviour. For the [getPlanFromEvents](https://xstate.js.org/docs/packages/xstate-test/#testmodel-getplanfromevents-events-options) method I wasn't able to pass down any parameters alongside the event type as part of the `events` array argument.  The [EventObject](https://github.com/statelyai/xstate/blob/main/packages/core/src/types.ts#L18) only accepts a `type` property. which prevents additional arguments that events typically include like `data` for services.

I might be doing something wrong here but wondering if this is a bug? If not what's the right way to approach this? Any examples or documentation around this would be great, thanks!

**Comment by RonHouben** [comment]
Heya,

You can do this by using the 'cases' option. 
The values you pass in there will be passed to the event.

Example:
```
createModel(myMachine).withEvents({
  TYPE: {
	  exec: ({ getByTestId }, event) => {
		  fireEvent.change(getByTestId('input'), { target: { value: event.value } });
	  },
	  cases: [{ value: 'my input text' }]
  }
})
```
